added husky, lint-staged

terazumova opened the pull request.

terazumova committed:
added husky, lint-staged

terazumova committed:
Merge branch 'master' into feature/MIC-12-add-husky-lint-staged

NooNoo1337 commented on .stylelintrc.json:
```diff
@@ -1,7 +1,7 @@
   "plugins": ["stylelint-prettier"],
   "rules": {
-    "prettier/prettier": true
+    "prettier/prettier": [true, { "endOfLine": "crlf", "printWidth": 120, "singleQuote": true, "tabWidth": 2 }]
```
https://github.com/prettier/stylelint-prettier#options

NooNoo1337 commented on package.json:
```diff
@@ -27,7 +27,13 @@
+    "style:lint": "stylelint src/**/*.css",
+    "js:lint": "eslint ./src/",
+    "lint:types": "tsc --noEmit",
+    "prettier:check": "prettier --config ./.prettierrc.json --check src/**/*",
```
**nitpick:** Если хранишь в корне проекта конфиг, его не надо явно передавать. И укажи форматы файлов (--check), которые явно подлежат проверке, а не все

NooNoo1337 commented on package.json:
```diff
@@ -27,7 +27,13 @@
+    "lint:types": "tsc --noEmit",
+    "prettier:check": "prettier --config ./.prettierrc.json --check src/**/*",
+    "prettier:write": "prettier --config ./.prettierrc.json --write src/**/*",
+    "prepare": "husky install"
```
**question:** Зачем?

terazumova committed:
fixing issues

terazumova committed:
fixing issues

terazumova committed:
fixing issues

terazumova committed:
fixing issues

terazumova committed:
fixed issues

terazumova committed:
fixed issues

terazumova committed:
fixed issues

terazumova committed:
fixed issues

terazumova committed:
fixed issues

terazumova committed:
fixed issues

terazumova committed:
fixed issues

terazumova committed:
fixed issues

terazumova committed:
fixed issues

terazumova committed:
fixed issues

terazumova committed:
fixed issues

terazumova committed:
fixed issues

fixed issues

fixed issues

fixed issues

fixed issues

fixed issues

fixed issues

fixed issues

fixed issues

terazumova committed:
removing extra script
